docs(hmr): document hmr.server option (fix #3587) (#3590)
vitejs/vite: docs/config/index.md

### server.hmr

- **Type:** `boolean | { protocol?: string, host?: string, port?: number, path?: string, timeout?: number, overlay?: boolean, clientPort?: number, server?: Server }`

  Disable or configure HMR connection (in cases where the HMR websocket must use a different address from the http server).

  Set `server.hmr.overlay` to `false` to disable the server error overlay.

>   When using `server.middlewareMode` and `server.https`, setting `server.hmr.server` to your HTTPS server will process HMR secure connection requests through your server.  This is of particular importance when using self-signed certificates.

Reviewer: ```suggestion
  When using `server.middlewareMode` and `server.https`, setting `server.hmr.server` to your HTTPS server will process HMR secure connection requests through your server. This can be helpful when using self-signed certificates.
```

Aside from this, maybe let's move this section underneath the `clientPort` so it aligns with the configs order

Author: OK.  Changes made as you suggested.
